packages/connect/src/api/firmware/uploadFirmware.ts

```diff
@@ -36,10 +38,31 @@ export const uploadFirmware = async (
     if (device.features.major_version === 1) {
         postConfirmationMessage(device);
         await typedCall('FirmwareErase', 'Success', {});
+
         postProgressMessage(device, 0, postMessage);
-        const { message } = await typedCall('FirmwareUpload', 'Success', {
+
+        let done: boolean = false;
+        let message = typedCall('FirmwareUpload', 'Success', {
             payload,
+        }).then(() => {
+            done = true;
         });
+
+        let i = 0;
+
+        const postProgressMessageIter = async () => {
+            i++;
+            postProgressMessage(device, Math.min(i * 2, 99), postMessage);
+            await createTimeoutPromise(300);
+            if (!done) {
+                postProgressMessageIter();
+            }
+        };
+
+        postProgressMessageIter();
+
+        await message;
```

Comment:
If you decide to keep the fake progress code ([this conversation](https://github.com/trezor/trezor-suite/pull/11300/files#r1545719604)), wouldn't this be a bit simpler? Also, I've changed `then` to `finally` so the fake progress is stopped even in case of failure.
```suggestion
        let i = 0;
        const progressTimer = setInterval(() => {
            i++;
            postProgressMessage(device, Math.min(i * 2, 99), postMessage);
        }, 300);

        const message = await typedCall('FirmwareUpload', 'Success', {
            payload,
        }).finally(() => {
            clearInterval(progressTimer);
        });        
```